lib/bindings/python/rust/lib.rs

```diff
@@ -428,6 +426,15 @@ impl DistributedRuntime {
         })
     }
 
+    /// Set the health status for an endpoint
+    fn set_endpoint_health_status(&self, endpoint: String, status: String) -> PyResult<()> {
```

Comment:
we should converge the handling for engine checks to be the same across backends - so either we set the endpoint heal;th status or exit - and then the status is implicit

may also simplify this to set overall status instead of endpoint by endpoint